app/utils/lib/getAvatarUrlFromUsername.js

```diff
@@ -1,21 +1,17 @@
 import { Session } from 'meteor/session';
 import { settings } from '../../settings';
+import { getAvatarURL } from './getAvatarURL';
 
 export const getAvatarUrlFromUsername = function(username) {
 	const externalSource = (settings.get('Accounts_AvatarExternalProviderUrl') || '').trim().replace(/\/$/, '');
 	if (externalSource !== '') {
 		return externalSource.replace('{username}', username);
 	}
-	const key = `avatar_random_${ username }`;
-	const random = typeof Session !== 'undefined' && typeof Session.keys[key] !== 'undefined' ? Session.keys[key] : 0;
 	if (username == null) {
 		return;
 	}
-	const cdnPrefix = (settings.get('CDN_PREFIX') || '').trim().replace(/\/$/, '');
-	const pathPrefix = (__meteor_runtime_config__.ROOT_URL_PATH_PREFIX || '').trim().replace(/\/$/, '');
-	let path = pathPrefix;
-	if (cdnPrefix) {
-		path = cdnPrefix + pathPrefix;
-	}
-	return `${ path }/avatar/${ encodeURIComponent(username) }?_dc=${ random }`;
+	const key = `avatar_random_${ username }`;
+	const random = typeof Session !== 'undefined' && typeof Session.keys[key] !== 'undefined' ? Session.keys[key] : 0;
+
+	return getAvatarURL(`${ encodeURIComponent(username) }?_dc=${ random }`);
```

Comment:
I've a few suggestions for the `getAvatarURL()` helper:

- use `encodeURIComponent()` inside the helper;
- add a parameter to optionally disable cache busting (appending `?_dc=${ randomKey }`).